src/iqm/qiskit_iqm/iqm_backend.py

```diff
@@ -78,18 +41,13 @@ class IQMBackendBase(BackendV2, ABC):
 
     def __init__(
         self,
-        architecture: Union[QuantumArchitectureSpecification, DynamicQuantumArchitecture],
+        architecture: DynamicQuantumArchitecture,
```

Comment:
I think this will be problematic for people running Qiskit on IQM with on devices with older software. We should probably support both.